raycicmd/converter.go

```diff
@@ -110,71 +112,68 @@ func parseStepEnvs(v any) ([]*envEntry, error) {
 	return entries, nil
 }
 
-func (c *converter) convertPipelineStep(step map[string]any) (
-	map[string]any, error,
-) {
-	if _, ok := step["wait"]; ok {
-		// a wait step
-		if err := checkStepKeys(step, waitStepAllowedKeys); err != nil {
-			return nil, fmt.Errorf("check wait step keys: %w", err)
-		}
-		return cloneMapExcept(step, waitStepDropKeys), nil
+func (c *converter) convertWait(step map[string]any) (map[string]any, error) {
+	// a wait step
+	if err := checkStepKeys(step, waitStepAllowedKeys); err != nil {
+		return nil, fmt.Errorf("check wait step keys: %w", err)
 	}
-	// special steps for building container images.
-	if _, ok := step["wanda"]; ok {
-		// a wanda step
-		if err := checkStepKeys(step, wandaStepAllowedKeys); err != nil {
-			return nil, fmt.Errorf("check wanda step keys: %w", err)
-		}
-		name, ok := stringInMap(step, "name")
-		if !ok {
-			return nil, fmt.Errorf("wanda step missing name")
-		}
-		file, ok := stringInMap(step, "wanda")
-		if !ok {
-			return nil, fmt.Errorf("wanda step file is not a string")
-		}
-		label, _ := stringInMap(step, "label")
-		instanceType, _ := stringInMap(step, "instance_type")
+	return cloneMapExcept(step, waitStepDropKeys), nil
+}
 
-		var matrix any
-		if m, ok := step["matrix"]; ok {
-			matrix = m
-		}
+func (c *converter) convertWanda(step map[string]any) (map[string]any, error) {
```

Comment:
this is also just splitting the wanda conversion logic out into its own function.